### ISSUE TITLE
Keras needs a flow_from_csv function

### ISSUE THREAD

**Issue by Vijayabhaskar96** [keras-team/keras#10069]
Keras has this awesome flow_from_directory function to read images that are stored in different directories named with their class names for image classification tasks, but it doesn't provide anything for regression tasks and also many datasets available on the internet simply has all the images stored in a single directory, with a csv file which maps the filename to the class names. So my suggestion is If someone can implement a flow_from_csv function that takes a CSV file as input. Something that looks like the below code will look great!
`datagen.flow_from_csv(csv="train.csv",filename_col="filenames",
output_cols=["value1","value2"],
sep=",",
target_size=(224, 224),
batch_size=32)`
The filename_col specifies the column that contains the filenames,
The output_cols (is a list) specifies the columns that will be treated as Y values (specifying a column that contains filenames here should also read the images and return them as Y).
The sep tells what seperator is used to seperate different columns(default ",")
target_size and batch_size has the same functions as in flow_from_directory.

**Comment by Dref360** [comment]
Feel free to submit a PR, but I don't know how flexible this API would be. We could get a ton of requests to handle different CSV format.

**Comment by minimaxir** [comment]
The real problem with this approach is that the input would be non-random, which will introduce a bias into the model.

**Comment by minimaxir** [comment]
The "shuffle happens only once" is the exact problem. See [this Stack Exchange](https://stats.stackexchange.com/questions/245502/shuffling-data-in-the-mini-batch-training-of-neural-network) discussion on why data should be shuffled after each epoch.

No, shuffling each CSV on load doesn't count. No, shuffling each CSV on load *and they order in which they are loaded* doesn't count either.

**Comment by mikelane** [comment]
I wasn’t clear, I apologize. The way my flow_from_csv works does in fact shuffle the data set for each epoch. My qualification above was that the data wasn’t being reloaded and reshuffled every time a mini batch was being yielded.

**Comment by Vijayabhaskar96** [comment]
Have a look at this [PR](https://github.com/keras-team/keras/pull/10127). This works exactly like flow_from_directory but with a dataframe, if shuffling is the problem,then it already exist in both `flow_from_directory` and `flow`
